packages/button/src/styledComponents/Button.js

```diff
@@ -105,10 +105,11 @@ function neutralStyle(props) {
 }
 
 function typeStyle(props) {
+  const theme = Object.assign({}, colorTypes, props.theme);
```

Comment:
Why not `spread operator` as you used on line 110? :P

`const theme = { ...colorTypes, ...props.theme };`

Reply:
Because `props.theme` can be `undefined`

Reply:
What if you set default `props.theme` to `{}`?

I personally prefer to use the spread operators, but if you guys prefer the `Object.assign` no problem

Reply:
Yes! I prefer spread operators, too.
It didn't come to me that styled components could have defaultProps. I'll change it to spread.